sdcm/cluster.py

```diff
@@ -3544,27 +3544,33 @@ def cql_connection_patient_exclusive(self, node, keyspace=None,
         return self.cql_connection_exclusive(**kwargs)
 
     def get_non_system_ks_cf_list(self, db_node,  # pylint: disable=too-many-arguments
-                                  filter_out_table_with_counter=False, filter_out_mv=False, filter_empty_tables=True) -> List[str]:
+                                  filter_out_table_with_counter=False, filter_out_mv=False, filter_empty_tables=True,
+                                  filter_by_keyspace: list = None) -> List[str]:
         return self.get_any_ks_cf_list(db_node, filter_out_table_with_counter=filter_out_table_with_counter,
                                        filter_out_mv=filter_out_mv, filter_empty_tables=filter_empty_tables,
-                                       filter_out_system=True, filter_out_cdc_log_tables=True)
+                                       filter_out_system=True, filter_out_cdc_log_tables=True, filter_by_keyspace=filter_by_keyspace)
```

Comment:
i would prefer not to change the methods of `get_non_system_ks_cf_list` and `get_any_ks_cf_list` by adding a new parameter of `filter_by_keyspace`.
It'll be better to add a new method that uses these methods and filtering it separately.

Reply:
Why do you prefer it?
The problem here - columns table may include a lot of rows. By adding filter by keyspace (that is pk) we can get smaller answer and faster.